libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1065,31 +1048,33 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 	return { std::move(ret) };
 }
 
+void StandardCompiler::configure(InputsAndSettings const& _inputsAndSettings, CompilerStack& _compilerStack)
+{
+	_compilerStack.setSources(_inputsAndSettings.sources);
+	for (auto const& smtLib2Response: _inputsAndSettings.smtLib2Responses)
+		_compilerStack.addSMTLib2Response(smtLib2Response.first, smtLib2Response.second);
+	_compilerStack.setViaIR(_inputsAndSettings.viaIR);
+	_compilerStack.setEVMVersion(_inputsAndSettings.evmVersion);
+	_compilerStack.setParserErrorRecovery(_inputsAndSettings.parserErrorRecovery);
+	_compilerStack.setRemappings(move(_inputsAndSettings.remappings));
+	_compilerStack.setOptimiserSettings(std::move(_inputsAndSettings.optimiserSettings));
+	_compilerStack.setRevertStringBehaviour(_inputsAndSettings.revertStrings);
+	_compilerStack.setLibraries(_inputsAndSettings.libraries);
+	_compilerStack.useMetadataLiteralSources(_inputsAndSettings.metadataLiteralSources);
+	_compilerStack.setMetadataHash(_inputsAndSettings.metadataHash);
+	_compilerStack.setRequestedContractNames(requestedContractNames(_inputsAndSettings.outputSelection));
+	_compilerStack.setModelCheckerSettings(_inputsAndSettings.modelCheckerSettings);
+
+	_compilerStack.enableEvmBytecodeGeneration(isEvmBytecodeRequested(_inputsAndSettings.outputSelection));
+	_compilerStack.enableIRGeneration(isIRRequested(_inputsAndSettings.outputSelection));
+	_compilerStack.enableEwasmGeneration(isEwasmRequested(_inputsAndSettings.outputSelection));
+}
```

Comment:
This function currently ignores `InputsAndSettings.language`. It was fine when the code was a part of `compileSolidity()` but now it's more generic, we should at least assert that the language is Solidity.

Or, better, it should create either `CompilerStack` or `AssemblyStack` based on the language. Or we should have two separate functions for that (e.g. `configureForSoldity()` and `conigureForYul()`), which might be more convenient - we'd  avoid having to return a variant that way.

Reply:
@cameel sooo many possibilities. :D

Reply:
Let's at least add an assert :)

BTW, I see that this code disappeared from the PR. Have you moved it to a separate one?